Add docs

ajeddeloh opened the pull request:
cc @bgilbert @arithx

arithx commented on README.md:
```diff
@@ -15,27 +20,27 @@ versions of the base and distro components that compose it. However a major
-base/
+`base/`
   Contains distro-agnostic code. Each package here targets only one Ignition
   spec verions.
```
s,verions,version,g

arithx commented on README.md:
```diff
@@ -15,27 +20,27 @@ versions of the base and distro components that compose it. However a major
 config version.
 
-internal/
+`internal/`
```
It seems that changing these from bullets has removed the line-break + tab on the descriptions.

ajeddeloh replied:
While that wasn't intentional, I think I'm ok with how it renders.

arithx commented on docs/examples.md:
````diff
@@ -0,0 +1,184 @@
+        - key1
+```
+
+This example modifies the existing `core` user, giving it a known password hash (this will enable login via password), and setting its ssh key.
````
The Ignition docs have the description of the example above the example, should we follow that pattern?

arithx commented on docs/examples.md:
````diff
@@ -0,0 +1,184 @@
+        - key3
+```
+
+This example will create two users, `user1` and `user2`. The first user has a password set and two ssh public keys authorized to log in as the user. The second user doesn't have a password set (so log in via password will be disabled), but have one ssh key.
````
s,but have one ssh key,but has one ssh key,g

arithx replied:
Might read better to use the actual usernames instead of first user or second user.

arithx commented on docs/examples.md:
````diff
@@ -0,0 +1,184 @@
+        id: 501
+```
+
+This example creates a file at `/opt/file` with the contents `Hello, world!`, permissions 0644 (so readable and writable by the owner, and only readable by everyone else), and the file is owned by user uid 500 and gid 501.
````
The example says `/opt/file` but the snippet has `/opt/file1`

arithx commented on docs/examples.md:
````diff
@@ -0,0 +1,184 @@
+            Environment="ETCD_NAME=infra0"
+```
+
+This example adds a drop-in for the `etcd-member` unit, setting the name for etcd to `infra0` with an environment variable. More information on systemd dropins can be found in [the docs][dropins].
````
Would `via` work better than `with` in the `with an environment variable` line?

ajeddeloh replied:
changing to `using`

arithx commented on docs/getting-started.md:
```diff
@@ -0,0 +1,37 @@
+# Getting started
+
+fcct is a tool that will consume a Fedora CoreOS Config and produce a JSON file that can be given to a Fedora CoreOS machine when it first boots to set the machine up. Using this config, a machine can be told to create users, create filesystems, set up the network, install systemd units, and more.
```
Maybe backtick wrap `fcct`?

arithx replied:
Maybe `configure the machine` instead of `set the machine up`?

arithx commented on docs/getting-started.md:
```diff
@@ -0,0 +1,37 @@
+
+To see some examples for what else ct can do, head over to the [examples][3].
+
+[1]: configuration-v1_0.md
```
Minor: use named anchors over numbered ones for ease of maintenance later on when potentially referencing the same section

arithx commented on docs/getting-started.md:
```diff
@@ -0,0 +1,37 @@
+
+Fedora CoreOS Configs are YAML files conforming to fcct's schema. For more information on the schema, take a look at [doc/configuration-v1_0.md][1].
+
+As a simple example, let's use ct to set the authorized ssh key for the core user on a Fedora CoreOS machine.
```
s,ct,fcct,g

arithx commented on docs/getting-started.md:
````diff
@@ -0,0 +1,37 @@
+{"ignition":{"config":{"replace":{"source":null,"verification":{}}},"security":{"tls":{}},"timeouts":{},"version":"3.0.0"},"passwd":{"users":[{"name":"core","sshAuthorizedKeys":["ssh-rsa ssh-rsa AAAAB3NzaC1yc..."]}]},"storage":{},"systemd":{}}
+```
+
+We can see that it produces a JSON file. This file isn't intended to be human-friendly, and will definitely be a pain to read/edit (especially if you have multi-line things like systemd units). Luckily, you shouldn't have to care about this file! Just provide it to a booting Fedora CoreOS machine and Ignition, the utility inside of Fedora CoreOS that receives this file, will know what to do with  ...
````
Worth turning Ignition into a link to the repo?

arithx commented on docs/getting-started.md:
```diff
@@ -0,0 +1,37 @@
+
+The method by which this file is provided to a Fedora CoreOS machine depends on the environment in which the machine is running. For instructions on a given provider, head over to the [list of supported platforms for Ignition][2].
+
+To see some examples for what else ct can do, head over to the [examples][3].
```
s,ct,fcct,g

bgilbert commented on docs/configuration-v1_0.md:
```diff
@@ -0,0 +1,125 @@
+The Fedora CoreOS configuration is a YAML document conforming to the following specification, with **_italicized_** entries being optional:
+
+* **variant** (string): must be `fcos`
+* **version** (string): the sematic version of the spec for this document. This document is for version `1.0.0` and generates Ignition configs with version `3.0.0`.
```
`semantic`

bgilbert commented on docs/configuration-v1_0.md:
```diff
@@ -0,0 +1,125 @@
+
+The Fedora CoreOS configuration is a YAML document conforming to the following specification, with **_italicized_** entries being optional:
+
+* **variant** (string): must be `fcos`
```
Trailing period for consistency

bgilbert commented on docs/getting-started.md:
```diff
@@ -0,0 +1,38 @@
+
+Fedora CoreOS Configs are YAML files conforming to `fcct`'s schema. For more information on the schema, take a look at [doc/configuration-v1_0.md][spec].
+
+As a simple example, let's use `fcct` to set the authorized ssh key for the core user on a Fedora CoreOS machine.
```
`core` user

bgilbert commented on docs/getting-started.md:
````diff
@@ -0,0 +1,38 @@
+
+As a simple example, let's use `fcct` to set the authorized ssh key for the core user on a Fedora CoreOS machine.
+
+```yaml
````
`yaml fedora-coreos-config`

bgilbert commented on docs/getting-started.md:
```diff
@@ -0,0 +1,38 @@
+
+In this above file, you'll want to set the `ssh-rsa AAAAB3NzaC1yc...` line to be your ssh public key (which is probably the contents of `~/.ssh/id_rsa.pub`, if you're on Linux).
+
+If we take this file and give it to `fcct`:
```
Should we start by explaining how to download it?

bgilbert commented on docs/getting-started.md:
```diff
@@ -0,0 +1,38 @@
+# Getting started
+
+`fcct` is a tool that will consume a Fedora CoreOS Config and produce a JSON file that can be given to a Fedora CoreOS machine when it first boots to configure the machine. Using this config, a machine can be told to create users, create filesystems, set up the network, install systemd units, and more.
```
Maybe: "`fcct`, the Fedora CoreOS Config Transpiler, is a tool that consumes a Fedora CoreOS Config and produces an Ignition config, which is a JSON document that can be given to a Fedora CoreOS machine to configure the machine when it first boots."

A machine being `told to` do things is not declarative, but never mind.

bgilbert commented on docs/getting-started.md:
```diff
@@ -0,0 +1,38 @@
+
+The method by which this file is provided to a Fedora CoreOS machine depends on the environment in which the machine is running. For instructions on a given provider, head over to the [list of supported platforms for Ignition][supported-platforms].
+
+To see some examples for what else `ct` can do, head over to the [examples][examples].
```
`fcct`

bgilbert commented on docs/examples.md:
```diff
@@ -0,0 +1,184 @@
+# Examples
+
+Here you can find a bunch of simple examples for using fcct, with some explanations about what they do. The examples here are in no way comprehensive, for a full list of all the options present in ct check out the [configuration specification][spec].
```
`present in fcct`

bgilbert commented on docs/examples.md:
````diff
@@ -0,0 +1,184 @@
+        - key1
+```
+
+This example will create two users, `user1` and `user2`. The `user1` has a password set and two ssh public keys authorized to log in as the user. `user2` doesn't have a password set (so log in via password will be disabled), but has one ssh key.
````
`login via password`

bgilbert commented on docs/examples.md:
```diff
@@ -0,0 +1,184 @@
+
+## Users and groups
+
+This example modifies the existing `core` user, giving it a known password hash (this will enable login via password), and setting its ssh key.
```
We're disabling password auth by default (https://github.com/coreos/fedora-coreos-tracker/issues/138) so these won't actually work.

bgilbert commented on docs/examples.md:
```diff
@@ -0,0 +1,184 @@
+version: 1.0.0
+systemd:
+  units:
+    - name: etcd-member.service
```
Let's use a service we ship in the OS.

ajeddeloh commented:
Fixed, added blurb on downloading/verifying fcct (although there's no releases to download yet), dropped everything related to password login, and swapped the example dropin to something useful (autologin on ttyS0)

bgilbert commented on docs/examples.md:
```diff
@@ -0,0 +1,147 @@
+        - name: autologin.conf
+          contents: |
+            [Service]
+	    TTYVTDisallocate=no
```
Bad indentation.

ajeddeloh commented:
Fixed
